core/src/main/java/hudson/model/UpdateSite.java

```diff
@@ -1243,6 +1248,43 @@ public boolean isNeededDependenciesCompatibleWithInstalledVersion(PluginManager.
             });
         }
 
+        /**
+         * Get the list of incompatible dependencies (if there are any, as determined by isNeededDependenciesCompatibleWithInstalledVersion)
+         *
+         * @since TODO
+         */
+        public List<Plugin> getDependenciesIncompatibleWithInstalledVersion() {
```

Comment:
Why not always use the version that requires a cache?

Reply:
Tbh I just implemented it like in e.g. `isNeededDependenciesCompatibleWithInstalledVersion()` and `isNeededDependenciesForNewerJenkins()`, because I thought the versions of the methods without parameters would be necessary. But I can of course remove this method, if they are not needed?

Reply:
If you add a new method, you create public API visible to plugins to be maintained forever. So if the goal is not to use it from plugins, I'd suggest to remove it.

Reply:
Okay, thank you for your explanation!